TFT/src/User/API/ScreenShot.c

```diff
@@ -0,0 +1,132 @@
+#include "ScreenShot.h"
+#include "includes.h"
+
+#if defined (SCREEN_SHOT_TO_SD)
+
+/* BMP file header 14byte */
+typedef struct __attribute__((__packed__))
+{
+	uint16_t bfType;           /* Must be "BM" (0x4D42), means .BMP file*/
+	uint32_t bfSize;           /* Bmp file sizes (byte) */
+	uint16_t bfReserved1;       /* Reserved 0, must be 0 */
+	uint16_t bfReserved2;       /* Reserved 1, must be 0 */
+	uint32_t bfoffBits;        /* The offset of the color pixel value (byte) */
+} BMP_FILEHEADER;
+
+/* BMP info header 40byte */
+typedef struct __attribute__((__packed__))
+{
+	uint32_t biSize;           /* Size of BMP_INFOHEADER */
+	uint32_t biWidth;          /* Bmp file width (pixel) */
+	uint32_t biHeight;         /* Bmp file height (pixel) */
+	uint16_t biPlanes;         /* Must be 1 */
+	uint16_t biBitCount;       /* Bit depth, 1/4/8/16/24/32... */
+	uint32_t biCompress;       /* Compression algorithm, 0 = Uncompressed */
+	uint32_t biSizeImage;      /* Image size(byte), can set to 0 when bmp Uncompressed */
+	uint32_t biXPelsPerMeter;  /* Horizontal resolution(pixels/meter) */
+	uint32_t biYPelsPerMeter;  /* Vertical resolution(pixels/meter) */
+	uint32_t biClrUsed;        /* Image used color number */
+	uint32_t biClrImportant;   /* Important color number */
+}BMP_INFOHEADER;  
+
+#define ALIGNMENT_4BYTE(n) (((n + 3) >> 2) << 2) //
+
+bool screenShotBMP(char *bmp)
+{
+  BMP_FILEHEADER bmp_file = {
+    .bfType = 0x4D42, // "BM"
+    .bfSize = (ALIGNMENT_4BYTE(LCD_WIDTH)  * LCD_HEIGHT) * 3 + sizeof(BMP_FILEHEADER) + sizeof(BMP_INFOHEADER),
+    .bfReserved1 = 0,
+    .bfReserved2 = 0,
+    .bfoffBits = sizeof(BMP_FILEHEADER) + sizeof(BMP_INFOHEADER),
+    };
+
+  BMP_INFOHEADER bmp_info = {
+    .biSize =  sizeof(BMP_INFOHEADER), // 40byte
+    .biWidth = ALIGNMENT_4BYTE(LCD_WIDTH),
+    .biHeight = LCD_HEIGHT,
+    .biPlanes = 1,
+    .biBitCount = 24, // 24bit depth
+    .biCompress = 0,
+    .biSizeImage = (ALIGNMENT_4BYTE(LCD_WIDTH)  * LCD_HEIGHT) * 3,
+    .biXPelsPerMeter = 0x1EC2,
+    .biYPelsPerMeter = 0x1EC2,
+    .biClrUsed = 0,
+    .biClrImportant = 0,
+    };
+
+  FIL   bmpFile;
+  UINT  mybw;
+
+  GUI_COLOR pix;
+
+  if(f_open(&bmpFile, bmp, FA_CREATE_NEW | FA_WRITE) != FR_OK)
```

Comment:
You could use  FA_CREATE_ALWAYS instead of FA_CREATE_NEW

Reply:
Hello, `FA_CREATE_ALWAYS` will cover the previous file. Currently, the menu title is used for the screenshot file name. When more than one screenshot is required in the same menu (especially 12864 Marlin mode), `FA_CREATE_ALWAYS` is not applicable. Therefore, I use `FA_CREATE_NEW` to judge whether the creation is successful or not. If the creation fails, the file name suffix will +1 and create again. The same menu is limited to 10 screen shots at most

Reply:
> If the creation fails, the file name suffix will +1 and create again. The same menu is limited to 10 screen shots at most

I missed this part, I thought only one screenshot per menu.